stats/kernel-selftests

```diff
@@ -730,39 +730,51 @@ class ResctrlStater < Stater
       @test_script = $1
     when /# Pass: Check\s+(.*)/
       # Pass: Check MBM diff within 5%
-
-      # The following line is repeated 3 times
-      # Pass: Check resctrl mountpoint "/sys/fs/resctrl" exists
-      stats.add "#{@test}.#{$1}", 'pass' unless stats.key? "#{@test}.#{$1}"
+      parse_cat_pass($1, stats)
     when /# SKIP\s+(.*)/
-      # The following line is repeated 3 times
-      # ok 4 # SKIP Hardware does not support CAT or CAT is disabled
-      stats.add "#{@test}.#{@test_script}", 'skip' unless stats.key? "#{@test}.#{@test_script}"
+      # SKIP: Hardware does not support CAT or CAT is disabled
+      parse_cat_skip(@test_script, stats)
     when /^not ok\s+(.*)/
       # not ok CAT: cache miss rate within 4%
-      @test_script = $1
-
-      # not ok 3 CMT: test
-      @test_script = $1.split(' ', 2)[1] if %w{1 2 3 4}.include? $1.split(' ', 2)[0]
-
-      # The following line is repeated 3 times
-      # not ok MBM: diff within 300%
-      stats.add "#{@test}.#{@test_script}", 'fail' unless stats.key? "#{@test}.#{@test_script}"
+      parse_cat_not_ok($1, stats)
     when /^ok\s+(.*)/
       # ok CAT: test
       # ok writing benchmark parameters to resctrl FS
-      @test_script = $1
-
-      # ok 2 MBA: schemata change
-      @test_script = $1.split(' ', 2)[1] if %w{1 2 3 4}.include? $1.split(' ', 2)[0]
-
-      # The following line is repeated 3 times
-      # ok resctrl mountpoint "/sys/fs/resctrl" exists
-      stats.add "#{@test}.#{@test_script}", 'pass' unless stats.key? "#{@test}.#{@test_script}"
+      parse_cat_ok($1, stats)
     else
       super(line, stats)
     end
   end
+
+  def parse_cat_pass(test_script, stats)
+    # The following line is repeated 3 times
+    # Pass: Check resctrl mountpoint "/sys/fs/resctrl" exists
+    stats.add "#{@test}.#{test_script}", 'pass' unless stats.key? "#{@test}.#{test_script}"
+  end
+
+  def parse_cat_skip(test_script, stats)
+    # The following line is repeated 3 times
+    # ok 4 # SKIP Hardware does not support CAT or CAT is disabled
+    stats.add "#{@test}.#{test_script}", 'skip' unless stats.key? "#{@test}.#{test_script}"
+  end
+
+  def parse_cat_not_ok(test_script, stats)
+    # not ok 3 CMT: test
+    test_script = test_script.split(' ', 2)[1] if %w{1 2 3 4}.include? test_script.split(' ', 2)[0]
+
+    # The following line is repeated 3 times
+    # not ok MBM: diff within 300%
+    stats.add "#{@test}.#{test_script}", 'fail' unless stats.key? "#{@test}.#{test_script}"
+  end
+
+  def parse_cat_ok(test_script, stats)
```

Comment:
this method has duplication to parse_cat_not_ok, prefer the logic like
```
def parse_cat_result(result, test_script, stats)
   test_script = test_script.split(' ', 2)[1] if %w{1 2 3 4}.include? test_script.split(' ', 2)[0]
 
  add_cat_result(result, test_script, stats)
```
The caller is like
    when /^(ok)\s+(.*)/
      # ok CAT: test
      # ok writing benchmark parameters to resctrl FS
      parse_cat_result($1, $2, stats)
```